### PR TITLE
P27_Refactor_RoleModule refactored role module

### PR DESCRIPTION
**Pull request by ibrahimejupi614**
P27_Refactor_RoleModule refactored role module

### CODE REVIEW

File: Postieri/Controllers/RolesController.cs [hunk 1]
```diff
@@ -13,10 +13,11 @@ public class RolesController : ControllerBase
 
       
         private readonly DataContext _context;
-        public RolesController(DataContext context)
+        private readonly IRolesService _rolesService;
+        public RolesController(DataContext context, IRolesService rolesService)
         {
```
Comment (dceneje) [inline Postieri/Controllers/RolesController.cs]: `DataContext` goes to the service.

**Review by dceneje — CHANGES_REQUESTED** [review]

File: Postieri/Interfaces/IRolesService.cs [hunk 2]
```diff
@@ -4,8 +4,9 @@ namespace Postieri.Interfaces
...
     public interface IRolesService
     {
-        void AddRole(Role role);
-        public void UpdateRole(Role request);
-        public void DeleteRole(Guid id);
+        List<Role> GetRoles();
+        List<Role> AddRole(Role role);
+        List<Role> UpdateRole(Role request);
+        List<Role> DeleteRole(Guid id);
     }
```
Comment (dceneje) [inline Postieri/Interfaces/IRolesService.cs]: `AddRole`, `UpdateRole` and `DeleteRole` need to be `bool`. Be consistent with params naming though, `role` or `request` for every method role parameter. I also suggest creating `Exists` method to use when dealing with crud.

**Review by dceneje — CHANGES_REQUESTED** [review]

**Review by dceneje — CHANGES_REQUESTED** [review]
Create `Exists` method and use it on crud to verify if a role exists before adding a new one, updating or deleting

**Comment by ibrahimejupi614** [comment]
> Create `Exists` method and use it on crud to verify if a role exists before adding a new one, updating or deleting

Ok

**Review by dceneje — APPROVED** [review]

**Pull request merged**